kogito-benchmarks-framework/src/main/java/org/kie/kogito/benchmarks/framework/Commands.java

```diff
@@ -0,0 +1,484 @@
+        appendlnSection(whatIDidReport, String.join(" ", cmd));
+        long buildStarts = System.currentTimeMillis();
+        buildService.shutdown();
+        buildService.awaitTermination(30, TimeUnit.MINUTES);
+        long buildEnds = System.currentTimeMillis();
+        long buildTimeMs = buildEnds - buildStarts;
+
+        return new BuildResult(buildTimeMs, buildLogA, 0);
+    }
+
+    public static RunInfo startApp(App app, StringBuilder whatIDidReport) throws IOException, InterruptedException {
+        File appDir = app.getAppDir();
+        File runLogA = new File(appDir.getAbsolutePath() + File.separator + "logs" + File.separator + app.mavenCommands.name().toLowerCase() + "-run.log");
+        List<String> cmd = getRunCommand(app.mavenCommands.mvnCmds[1]);
+        appendln(whatIDidReport, appDir.getAbsolutePath());
+        appendlnSection(whatIDidReport, String.join(" ", cmd));
+        Process pA = runCommand(cmd, appDir, runLogA);
+        // Test web pages
+        long timeToFirstOKRequest = WebpageTester.testWeb(app.urlContent.urlContent[0][0], 10, app.urlContent.urlContent[0][1], true);
+        logger.info("Testing web page content...");
+        for (String[] urlContent : app.urlContent.urlContent) {
+            WebpageTester.testWeb(urlContent[0], 5, urlContent[1], false);
+        }
+
+        return new RunInfo(pA, runLogA, timeToFirstOKRequest);
+    }
+
+    public static void setCPUAffinity(RunInfo runInfo, int numOfCores) throws IOException, InterruptedException {
+        logger.info("Setting CPU affinity of app to " + numOfCores + " core(s)");
+        String listOfCores = "0-" + (numOfCores - 1);
+        String processId = String.valueOf(runInfo.getProcess().pid());
+        ProcessBuilder pBuilder = new ProcessBuilder("taskset", "-cpa", listOfCores, processId);
+        pBuilder.redirectErrorStream(true);
+        Process p = pBuilder.start();
+        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(p.getInputStream()))) {
+            String newCPUAffinity = bufferedReader.lines()
+                    .filter(s -> s.contains("new affinity list"))
+                    .findFirst()
+                    .orElseThrow(() -> new RuntimeException("New CPU Affinity not set!"));
+            String expectedAffinityList = numOfCores == 1 ? "0" : numOfCores == 2 ? "0,1" : listOfCores;
+            Assertions.assertThat(newCPUAffinity).contains(expectedAffinityList);
+            p.waitFor();
+        }
+        Assertions.assertThat(p.exitValue()).isEqualTo(0);
+    }
+
+    public static void cleanDirOrFile(String... path) {
+        for (String s : path) {
+            try {
+                Files.walk(Paths.get(s))
+                        .sorted(Comparator.reverseOrder())
+                        .map(Path::toFile)
+                        .forEach(File::delete);
+            } catch (IOException e) {
+                logger.warn("Unable to delete directories or files", e);
```

Comment:
isn't this a serious issue? Is it safe to just log?

Reply:
Originally there was something like [this](https://github.com/quarkus-qe/quarkus-startstop/blob/ddcf9f3ce5995ef1fee044d3f434c1881b3ab9a6/testsuite/src/it/java/io/quarkus/ts/startstop/utils/Commands.java#L227) :) I replaced it with throwing a RuntimeException and thus failing a single test.